app/controllers/application_controller.rb

```diff
@@ -77,6 +77,11 @@ def find_published_asset
     track_not_found unless @asset.published? || current_user_is_admin_or_owner?(@asset.user)
   end
 
+  def track_not_found
```

Comment:
Moved this method from assets_controller.rb since it's only used by two methods above.